### PR TITLE
Remove usage of metaprogramming in device helpers

### PR DESCRIPTION
**Pull request by lionel-**
To fix the R-devel warning. It seems more robust not to inline programmatically the arguments in generated functions so I've now hardcoded the implementations. For simplicity arguments are passed via `...` and the documentation now contains a list of links to wrapped functions so the relevant signatures can be browsed by following the links.

Does this seem reasonable enough @hadley ?

I was also expecting these notes to disappear following this change:

```
  pdf_dev: no visible binding for global variable ‘grDevices’
  pdf_dev: no visible binding for global variable ‘pdf’
  pdf_dev: no visible binding for global variable ‘dev.cur’
  png_dev: no visible binding for global variable ‘grDevices’
  png_dev: no visible binding for global variable ‘png’
  png_dev: no visible binding for global variable ‘dev.cur’
```

Somehow they're still there.

### CODE REVIEW

**Comment by hadley** [comment]
All the weird variable binding errors are caused by `on_load(`::` <- base::`::`)`

**Review by hadley — COMMENTED** [review]
This seems like a cleaner approach to me.

File: R/devices.R [hunk 1]
```diff
@@ -77,105 +33,217 @@ jpeg_dev <- function(filename, ...) {
...
+  defer(dev_close(old), envir = .local_envir)
+  invisible(old)
+}
+
+
+# Internal *_dev functions ------------------------------------------------
+
+pdf_dev <- function(filename, ...) {
+  grDevices::pdf(file = filename, ...)
+  grDevices::dev.cur()
```
[the remark this answers is not in the document]
Comment (lionel-) [reply R/devices.R]: Good idea. How does this now look? I've abstracted device closing in a `defer_dev_close()` helper.

**Comment by lionel-** [comment]
> All the weird variable binding errors are caused by on_load(:: <- base::::)

oh no :-(

**Pull request merged**